modules/core/src/main/scala/gem/math/Angle.scala

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.math
+
+import scalaz.Monoid
+
+/**
+ * Exact angles represented as integral milliarcseconds. These values form an Abelian group over
+ * addition, where the inverse is reflection around the 0-180° axis. The subgroup of angles where
+ * integral milliarcseconds correspond with clock milliseconds (i.e., where they are evenly
+ * divisible by 15 milliarcseconds) is represented by the HourAgle subtype.
+ * @param toMilliarcseconds This angle in milliarcseconds. Exact.
+ */
+sealed class Angle (val toMilliarcseconds: Int) {
+
+  // Sanity checks … should be correct via the companion constructor.
+  assert(toMilliarcseconds >= 0, s"Invariant violated. $toMilliarcseconds is negative.")
```

Comment:
The `fromXXX` methods guarantee these but I wanted a double-check while developing since I have been changing things so much. When we're sure it's right we can turn these off.